website/src/content/docs/docs/libraries/http/operations.md

```diff
@@ -580,6 +580,150 @@ model Thing {
+- `flavor` may or may not occur in the request, if it does occur, the resource value can be updated to one of the appropriate values. If set to `null` it is returned to its default value (`vanilla`)
+- `related` may or may not occur in the request, if it does occur, the resource value can be updated or erased (set to `{}`) by sending null.
+  - Since `related` is a keyed type (Object), each key/value pair in the request will be treated as follows:
+  - If the key exists in the 'related' field of the resource, the value is merged with the existing value.
+  - If the key does not exist in the 'related' field of the resource, the key/value pair is added.
+- `tags` may or may not occur in the request, if it does occur, the resource can be replaced by a new array, or erased by sending null.
+
+### The MergePatch Transform
+
+Generalizing these rules creates a type transform of the resource Model into a Model defining the corresponding Patch request body. The properties of a resource define the allowed properties and their types in a Patch request body for that resource as follows:
+
+- The `type` of a property indicates the allowed type of the corresponding property in a Patch request.
+- `Required` properties without a default value are _optional_ in a Patch request, but if present, the value cannot be null.
+- `Optional` properties are _optional_ in a Patch request, and, if present, can contain any valid value or null.
+- Properties with a default value are _optional_ in a Patch request, but have no default value in the request body. If present, they can contain any valid value or null. A _null_ value will not erase the property, but set it back to its default value.
+- `Model` and `Record` typed properties, when present in a Patch request are merged with the existing resource value in accordance with the merge-patch algorithm: this transformation is recursively applied to these values.
+- `Array` typed properties, when present in a Patch request replace the existing resource value in accordance with the merge-patch algorithm. This transformation is _not_ recursively applied to these values.
+- Properties required to deserialize a request (for example, discriminators) are _required_ in a Patch request body.
+
+### Constraints
+
+Because JSON merge-patch only describes the request body of an http request, the merge-patch
+transform in TypeSpec does not allow http envelope property metadata like `@query`, `@path`,
+`@header`, `@cookie`, and `@statusCode` in http requests.
+
+The merge-patch templates will **emit an error diagnostic** if a model containing Http metadata
+properties is passed in. For example:
+
+```tsp
+model ResourceWithMetadata {
+  @path id: string;
+  @header eTag: string;
+  description: string;
+}
+
+// THIS WILL RESULT IN AN ERROR
+@patch op update(...MergePatchUpdate<ResourceWithMetadata>): ResourceWithMetadata;
+```
```

Comment:
Perhaps worth mentioning also that it is invalid to pass a model into `MergePatchUpdate` or `MergePatchCreateOrUpdate` that contains `null` anywhere in its ordinary value space, since `null` has special meaning in JSON Merge Patch.